website/components/PropTable.tsx

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import { css } from 'emotion';
+import { Table, TableHeader, Row, Cell } from '@leafygreen-ui/table';
+import { Subtitle, InlineCode } from '@leafygreen-ui/typography/dist';
+import { OneOf } from '@leafygreen-ui/lib';
+import PropDefinition from 'components/PropDefinition';
+import TypographyPropTable from 'components/TypographyPropTable';
+
+const subtitleBottomMargin = css`
+  margin-bottom: 24px;
+`;
+
+const tableBottomMargin = css`
+  margin-bottom: 56px;
+`;
+
+const tableHeaderNames: Array<'prop' | 'type' | 'description' | 'default'> = [
+  'prop',
+  'type',
+  'description',
+  'default',
+];
+
+interface TableDataInterface {
+  prop: string;
+  type: string;
+  description: string;
+  default: string;
+}
+
+interface RowInterface {
+  children: Array<CellRoot>;
+}
+
+export interface CellRoot {
+  children: Array<CellChild>;
+}
+
+type CellChild = OneOf<
+  {
+    children: Array<CellLeaf>;
+  },
+  CellLeaf
+>;
+
+interface CellLeaf {
+  children?: undefined;
+  value: string;
+  type: string;
+  depth: number;
+}
+
+function getTableData(rows: Array<RowInterface>): Array<TableDataInterface> {
+  if (!rows) {
+    return null;
+  }
+
+  const rowMap = rows.map(row => {
+    const rowObj: Partial<TableDataInterface> = {};
+    row.children.map((cell, index) => {
+      const value =
+        cell.children
+          .map(child => {
+            if (child.children) {
+              return child.children.map(child => child.value).join('');
+            }
+
+            return child.value;
+          })
+          .join('') || '-';
+
+      rowObj[tableHeaderNames[index]] = value;
+    });
+    return rowObj;
+  });
+
+  rowMap.shift();
+
+  return rowMap as Array<TableDataInterface>;
+}
+
+function PropTable({
+  markdownAst,
+  component,
+}: {
+  markdownAst: CellRoot;
+  component: string;
+}) {
+  let peerDepIndex: number | undefined;
+
+  const headers = markdownAst.children
+    .filter(treeItem => treeItem.type === 'heading' && treeItem.depth === 1)
```

Comment:
this should be a hint that `ReadmeMarkdown.children` is a discriminated union on `type`. 
```tsx
interface ReadmeMarkdown {
  children: Heading | ...;
}

interface Heading {
  type: 'heading';
  ...
}
```

Reply:
👍